ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -706,6 +734,21 @@ func (r *RayJobReconciler) checkK8sJobAndUpdateStatusIfNeeded(ctx context.Contex
 				rayJob.Status.Reason = rayv1.SubmissionFailed
 				rayJob.Status.Message = fmt.Sprintf("Job submission has failed. Reason: %s. Message: %s", cond.Reason, cond.Message)
 			}
+
+			failedCount := int32(0)
+			if rayJob.Status.Failed != nil {
+				failedCount = *rayJob.Status.Failed
+			}
+			failedCount++
+			rayJob.Status.Failed = ptr.To[int32](failedCount)
+
+			if rayJob.Spec.BackoffLimit != nil && *rayJob.Spec.BackoffLimit > 0 &&
```

Comment:
We can also check whether BackoffLimit is smaller than 0 or not in `validateRayJobSpec`, and then we don't need to check `*rayJob.Spec.BackoffLimit > 0` here.

Reply:
done